src/main/java/uk/sky/cirrus/CqlMigratorImpl.java

```diff
@@ -0,0 +1,125 @@
+package uk.sky.cirrus;
+
+import com.datastax.driver.core.Cluster;
+import com.datastax.driver.core.ConsistencyLevel;
+import com.datastax.driver.core.QueryOptions;
+import com.datastax.driver.core.Session;
+import com.google.common.base.Preconditions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.sky.cirrus.locking.CassandraLockConfig;
+import uk.sky.cirrus.locking.CassandraLockingMechanism;
+import uk.sky.cirrus.locking.Lock;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.stream.Collectors;
+
+/**
+ * Standard implementation for {@code CqlMigrator}
+ *
+ * Implements Locking by providing mutex locking via the locks table.
+ * Client tries to acquire the lock and would only succeed for one thread.
+ * The inserted record will be deleted once the client releases the lock it holds, hence making way for a subsequent client thread to acquire the lock.
+ * Locking is done per client per keyspace. A client can obtain a lock on a different keyspace if no other client is holding a lock for that keyspace.
+ *
+ */
+
+public final class CqlMigratorImpl implements CqlMigrator {
```

Comment:
Hey guys, I don't get why we extracted an interface here.

So our public api is going to be 'new CqlMigratorImpl(lockConfig)'? I don't think we should have an 'Impl' class as our public API.

Can we either
- Remove the interface
- Introduce a factory method that returns the interface (guava has good patterns here to follow with object instantiation).

In addition, if we remove the interface, we should probably make our API through a static factory method which returns an instance rather than a raw constructor. We'll get a lot more flexibility with the former to change things later on without breaking the API (e.g. a #create() method).